src/js/stores/MesosLogStore.js

```diff
@@ -1,30 +1,62 @@
-import {Store} from 'mesosphere-shared-reactjs';
+import BaseStore from './BaseStore';
 
 import ActionTypes from '../constants/ActionTypes';
 import AppDispatcher from '../events/AppDispatcher';
 import Config from '../config/Config';
-import EventTypes from '../constants/EventTypes';
-import GetSetMixin from '../mixins/GetSetMixin';
+import EventTypes from '../constants/EventTypes';;
 import Item from '../structs/Item';
 import LogBuffer from '../structs/LogBuffer';
 import MesosLogActions from '../events/MesosLogActions';
 
 const MAX_FILE_SIZE = 50000;
 
-const MesosLogStore = Store.createStore({
-  storeID: 'mesosLog',
-
-  mixins: [GetSetMixin],
-
-  addChangeListener: function (eventName, callback) {
+class MesosLogStore extends BaseStore {
+
+  constructor() {
+    super(...arguments);
+
+    this.dispatcherIndex = AppDispatcher.register((payload) => {
```

Comment:
@aldipower If we wouldn't export an [instance](https://github.com/dcos/dcos-ui/pull/286/files/39bd1df7e5b21ad2d676aac226deed7ae360be09#diff-d2871a01da8a1ac9c58e9633e5e84505R203) but the class itself we would be in real danger of creating memory leaks as there is no unregister. 
We could introduce a new singleton pattern which makes sure that we will always have only one instance in memory even if we export the class. But thats a different story and not part of this PR.